frontend/app/components/form.js

```diff
@@ -0,0 +1,44 @@
+import { action } from '@ember/object';
+import { scheduleOnce } from '@ember/runloop';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { resolve } from 'rsvp';
+
+export default class FormComponent extends Component {
+  @tracked loading = false;
+  @tracked submitted = false;
+
+  constructor(...args) {
+    super(...args);
+
+    if (this.args.model && this.args.model.validate) {
+      scheduleOnce('actions', this, 'validateModel', this.args.model);
+    }
+  }
```

Comment:
I think the problem there is your empty changeset which is just an empty object. So there is no property to validate. If you pass an object with all possible properties (but with a `null` value) this should not be necessary

Reply:
With empty changeset i meant a changeset with validations and without a model passed to it
```js
import { Changeset } from 'ember-changeset';
import lookupValidator from 'ember-changeset-validations';
import SomeValidations from 'app/validations/some'

emptyChangeset = Changeset({}, lookupValidator(SomeValidations), SomeValidations)
```
This would not be an empty object and would look more like this:

```js
{
    "validationMap": {
        "someFieldToValidate": [
            null,
            null,
            null
        ],
        "someOtherFieldToValidate": [
            null,
            null,
            null
        ]
    },
    "__changeset__": "__CHANGESET__",
    "_eventedNotifiers": {},
    "_options": {
        "skipValidate": false
    },
    "_runningValidations": {},
    "_errorsCache": {}
}
```

Reply:
I think this was exactly the problem. Let's say your model looks like this:

```js
const model = {
  foo: 1,
  bar: 2
}
```

Now if you want to create a new one, you need an empty changeset that looks like this:

```js
const model = {
  foo: null,
  bar: null
}
```

Instead of just an empty object (`{}`).

If you do not have the validated properties defined, it won't have any properties to validate on the first validation run, which is why you need to validate in the constuctor for the validation to work..

Reply:
> I think the problem there is your empty changeset which is just an empty object. So there is no property to validate. If you pass an object with all possible properties (but with a null value) this should not be necessary

This does not actually change anything as described in [ember-changeset](https://github.com/poteto/ember-changeset#usage):
> The helper receives any Object (including DS.Model, Ember.Object, or even POJOs) and an optional validator action. If a validator is passed into the helper, the changeset will attempt to call that function **when a value changes**.

Therefore passing an Object with all properties as `null` doesn't work, since they are not initially validated.

With `scheduleOnce('actions', this, 'validateModel', this.args.model);` in the constructor we schedule for the changeset/model to [validate](https://github.com/poteto/ember-changeset#validate) all fields.

Reply:
As discussed, makes sense.